bridge-api/definitions/shared_module_metadata.yml

```diff
@@ -0,0 +1,55 @@
+type: object
+description: Metadata for shared modules in the shared module library.
+required:
+    - id
+    - name
+    - version
+properties:
+    id:
+        type: string
+        description: Unique ID of the module.
+    licenseRestricted:
+        type: boolean
+        description: True if usage of this module is restricted by license or copyright.
+    moduleType:
+        $ref: ./enums/shared_module_type.yml
+        readOnly: true
```

Comment:
It should be okay. On the server-side, moduleType is actually inferred from the presence/absence of schemaId and surveyGuid, so the client shouldn't be able to set it anyway.

Ironically, the readOnly flag is actually ignored by swagger for whatever reason, and there *is* in fact a setter for it on the JavaSDK...